android/KMEA/app/src/main/java/com/tavultesoft/kmea/packages/PackageProcessor.java

```diff
@@ -408,6 +410,52 @@ public boolean accept(File pathname) {
     return false;
   }
 
+  /**
+   * Get a list of available keyboard and language pairings that are available to add
+   * (parses the kmp.json for the languagae list)
+   * @param packageID - String of the package ID
+   * @param keyboardID - String of the keyboard ID
+   * @return List of keyboards that haven't been added yet
+   */
+  public List<Keyboard> getLanguageList(String packageID, String keyboardID) {
+    File packagePath = new File(resourceRoot, KMManager.KMDefault_AssetPackages + File.separator + packageID);
+    List<Keyboard> list = new ArrayList<Keyboard>();
+    JSONObject infoJSON = loadPackageInfo(packagePath);
+    try {
+      JSONArray keyboards = infoJSON.getJSONArray("keyboards");
+
+      for (int i = 0; i < keyboards.length(); i++) {
+        JSONObject keyboard = keyboards.getJSONObject(i);
+        if (keyboardID.equals(keyboard.getString("id"))) {
+          // Find the keyboard already installed. We'll use it as the base
+          // for creating new keyboards
+          int baseKeyboardIndex =  KeyboardController.getInstance().getKeyboardIndex(packageID, keyboardID, "");
+          if (baseKeyboardIndex == KeyboardController.INDEX_NOT_FOUND) {
+            continue;
+          }
+          Keyboard baseKeyboard = KeyboardController.getInstance().getKeyboardInfo(baseKeyboardIndex);
+
+          JSONArray languages = keyboard.getJSONArray("languages");
```

Comment:
Can we factor this into a separate function? It's just a little cleaner than nesting.